kweeuhree/plant-parent-redux
# [ISSUE 9] - Add useNavigateToPath Hook

<h3>Description</h3>

This pull request introduces a custom hook, `useNavigateToPath`, which allows developers to navigate to a specified path with an optional timeout for delayed navigation. The hook concatenates a base path with a given path and supports both immediate and delayed redirects. This implementation promotes code reusability and flexibility, making it easier to handle navigation throughout the app.

I chose this approach to streamline the redirection logic in our application, ensuring that any component requiring navigation with a potential delay can easily leverage this reusable hook. One key takeaway was learning how to handle navigation delays using React's `useNavigate` while maintaining simplicity and flexibility in the hook's API.

<h3>Related Issue</h3>

Closes #9 

<h3>Acceptance Criteria</h3>

- Implement a custom hook `useNavigateToPath`.
- Accept a `basePath` parameter that is concatenated with a given `path`.
- Expose `handleNavigateToPath`, which:
  - Navigates to the specified path immediately if no `timeout` is provided.
  - Delays navigation based on the provided `timeout` (in milliseconds).

<h3>Type of Changes</h3>

- Enhancement

<h3>Updates</h3>

<h3>Before</h3>

There was no reusable logic for handling navigation with an optional delay. Each component had to implement its own redirection logic or reuse the same code.

<h3>After</h3>

The `useNavigateToPath` hook encapsulates the navigation logic with a customizable timeout, making it reusable across different components.

<h3>Testing Steps / QA Criteria</h3>

1. Pull this branch and run the app.
2. Use the `useNavigateToPath` hook in any component, passing a base path.
3. Test navigation with both immediate redirect (without timeout) and delayed redirect (by passing a `timeout` value).
4. Ensure that the user is navigated to the correct path, with or without the delay, as expected.

For example, in a component:
```javascript
const navigate = useNavigateToPath('');
navigate('/profile', 1000); // Redirects to /profile after 1 second.
